src/mo_frontend/definedness.ml

```diff
@@ -192,9 +195,13 @@ and decs msgs decs : group =
     (d.at, defs, eager_vars f, delayed_vars f)
   ) decs
 
-and group msgs (grp : group) : f =
+and group msgs ?(extern=None) (grp : group) : f =
   (* Create a map from declared variable to their definition point *)
   let defWhen = M.disjoint_unions (List.mapi (fun i (_, defs, _, _) -> map_of_set i defs) grp) in
+  (* Insert the externally defined binding in front if present, non-shadowing *)
+  let defWhen = match extern with
+    | Some b when M.find_opt b.it defWhen |> Option.is_none -> M.add b.it (-1) defWhen
+    | _ -> defWhen in
```

Comment:
You might not need this if you just extend the decs appropriately,

Reply:
#4735 is merged, thanks!